src/arch/ia32e/interrupts/idt.rs

```diff
@@ -66,67 +73,81 @@ impl InterruptDescriptorTableRegister {
     }
 }
 
-fn setup_predefined(idt: &mut InterruptDescriptorTable) {
+static mut IDT: InterruptDescriptorTable = InterruptDescriptorTable::const_default();
+
+unsafe fn setup_predefined() {
     let kernel_segment =
         SegmentSelector::new(GDT_KERNEL_CODE, TableIndicator::GDT, PrivilegeLevel::Kernel);
 
     // Divide error
-    idt.entries[0] = Gate::new(div_by_zero as *const () as u64, kernel_segment).present(true);
+    IDT.entries[0] = Gate::new(div_by_zero as *const () as u64, kernel_segment).present(true).kind(Kind::Interrupt);
```

Comment:
We might introduce `Gate::interrupt` constructor instead to avoid this call. This constructor would set the gate kind to interrupt when called.

Reply:
Done in 2c07d2ce3a4d4de43c7feec62f5bf91787cb30b5.
Added `Gate::interrupt` and `Gate::trap` and made `Gate::new` private to prevent further confusion.